File: JavaFactorial/src/de/luschny/math/factorial/IFactorialFunction.java
// Copyright (C) 2004-2009 Peter Luschny, MIT License applies.
// See http://en.wikipedia.org/wiki/MIT_License
// Visit http://www.luschny.de/math/factorial/FastFactorialFunctions.htm
// Comments mail to: peter(at)luschny.de
package de.luschny.math.factorial;

import de.luschny.math.arithmetic.Xint;

// An interface for the factorial function
//        n! = 1*2*3*...*n
// for nonnegative integer values n.
public interface IFactorialFunction {

    Xint factorial(int n);

    String getName();
}

// endOfIFactorialFunction

File: SilverFactorial64/Sharith/Factorial/IFactorialFunction.cs
/// -------- ToujoursEnBeta
/// Author & Copyright : Peter Luschny
/// License: LGPL version 3.0 or (at your option)
/// Creative Commons Attribution-ShareAlike 3.0
/// Comments mail to: peter(at)luschny.de
/// Created: 2010-03-01

namespace Sharith.Math.Factorial 
{
    using XInt = Sharith.Arithmetic.XInt;

    /// <summary> An interface for the factorial function
    /// n! = 1*2*3*...*n
    /// for nonnegative integer values n.
    /// </summary>
    public interface IFactorialFunction
    {
        string Name { get; }
        XInt Factorial(int n);
    }

} // endOfIFactorialFunction
